### ISSUE TITLE
Credentials not found when using source_profile without specifying AWS_SDK_LOAD_CONFIG

### ISSUE THREAD

**Issue by ryandens** [awslabs/amazon-ecr-credential-helper#232]
Hello! I'm having some trouble taking advantage of some functionality added in the latest release regarding pulling credentials from an assumed role specified with `role_arn` and `source_profile`. I confirmed I'm on the latest release [v0.4.0](https://github.com/awslabs/amazon-ecr-credential-helper/releases/tag/v0.4.0) which mention the `source_profile` functionality in the release notes.
```bash
$ docker-credential-ecr-login -v
amazon-ecr-credential-helper
Version:    0.4.0
Git commit: 1a5791b
```

## Steps to reproduce

In `~/.aws/credentials`
```
[default]
aws_access_key_id = foo
aws_secret_access_key = bar
```

In `~/.aws/config`
```
[profile admin]
source_profile = default
role_arn = <admin-role-arn>
```

Assume the role by setting the `AWS_PROFILE` environment variable and then test, verify the role was assumed correctly using the AWS CLI, and then demonstrate the ECR credential helper not working. Also set the `AWS_SDK_LOAD_CONFIG=true` environment variable and it works
```
$ export AWS_PROFILE=admin
$ aws sts get-caller-identity
{
    "UserId": "<id>",
    "Account": "<Account>",
    "Arn": "<admin-role-arn>"
}
$ echo <Account>.dkr.ecr.us-east-1.amazonaws.com | docker-credential-ecr-login get
credentials not found in native keychain
$ echo <Account>.dkr.ecr.us-east-1.amazonaws.com | AWS_SDK_LOAD_CONFIG=true docker-credential-ecr-login get
{"ServerURL":" <Account>.dkr.ecr.us-east-1.amazonaws.com","Username":"AWS","Secret":"baz"}
```

## Source of issue
It seems shared config was enabled by default on February 3rd in [this commit](https://github.com/awslabs/amazon-ecr-credential-helper/commit/2389db6c15b92dbd3975267f96f76d2cc534ae1a). The documentation was also updated which makes sense.  However, there has not been a release since January 6th based on your [releases GitHub page](https://github.com/awslabs/amazon-ecr-credential-helper/releases). I understand software doesn't get released right away, but it took me quite a while to realize the documentation listed in this repository was not accurate for the latest release. I'd be happy to update the documentation to the effect of: 

> The Amazon ECR Docker Credential Helper reads and supports some configuration options specified in the AWS
shared configuration file (`~/.aws/config`).  On versions 0.5.X and higher, this is enabled by default. To disable these options, you must set the `AWS_SDK_LOAD_CONFIG` environment variable to `false` On versions 0.4.X and lower, this is disabled by default and you must set the  `AWS_SDK_LOAD_CONFIG` environment variable to `true` to take advantage of this functionality.

But, I understand including version-specific documentation in your README could snowball and may be prohibitively hard to maintain. Publishing versioned documentation would definitely be welcome, but that's a significant undertaking. If nothing else, I hope this GitHub issue helps people figure out why they need to specify `AWS_SDK_LOAD_CONFIG` despite documentation saying otherwise

**Comment by RulerOf** [comment]
Thanks for this analysis.

To be clear to anyone who wanders in here: **Installing from source fixes this problem**.

**Comment by TBBle** [comment]
I just bounced off this too. [Docker Desktop from 2.4.0.0 is now bundling the latest release of this helper](https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2400), so it'd be good to have a new release so that the bundled version with Docker Desktop has the fix for #198 by default.

For now, a feasible workaround is to set the env-var `AWS_SDK_LOAD_CONFIG=true`.

**Comment by samuelkarp** [comment]
[v0.5.0](https://github.com/awslabs/amazon-ecr-credential-helper/releases/tag/v0.5.0) removes the need to set `AWS_SDK_LOAD_CONFIG`.
